src/web3/merkleDistributor.ts

```diff
@@ -0,0 +1,80 @@
+import { BigNumber } from 'ethers';
+import { AbiItem } from 'web3-utils';
+import Web3Contract, { createAbiItem } from 'web3/web3Contract';
+
+import airdropData from '../merkle-distributor/airdrop.json';
+
+const ABI: AbiItem[] = [
+  createAbiItem('isClaimed', ['uint256'], ['bool']),
+  createAbiItem('claim', ['uint256', 'address', 'uint256', 'bytes32[]']),
+  createAbiItem('calculateAdjustedAmount', ['uint256'], ['uint256', 'uint256', 'uint256']),
+];
+
+export default class MerkleDistributor extends Web3Contract {
+  isAirdropClaimed?: boolean;
+  claimIndex?: number;
+  claimAmount?: string;
+  adjustedAmount?: string;
+
+  constructor(abi: AbiItem[], address: string) {
+    super([...ABI, ...abi], address, '');
+
+    this.on(Web3Contract.UPDATE_ACCOUNT, () => {
+      if (!this.account) {
+        this.isAirdropClaimed = false;
+        this.claimIndex = -1;
+        this.claimAmount = undefined;
+        this.adjustedAmount = undefined;
+        this.emit(Web3Contract.UPDATE_DATA);
+      }
+
+      const airdropAccounts = airdropData.map(drop => ({
+        account: drop.address,
+        amount: BigNumber.from(BigNumber.from(drop.earnings.toString())),
+      }));
+      this.claimIndex = airdropAccounts.findIndex(o => o.account === this.account);
+      this.claimIndex !== -1
+        ? (this.claimAmount = this.getClaimAmount(this.account || '', airdropData))
+        : (this.claimAmount = undefined);
+      this.adjustedAmount = undefined;
+    });
+  }
+
+  getClaimAmount(address: string, airdropData: any[]): string {
+    return airdropData.find(e => e.address === address).earnings;
```

Comment:
there is posibility the item not to be found, so

airdropData.find(e => e.address === address) will return undefined and '.earnings' will crash app,

would be better to have like this

```
getClaimAmount(address: string, airdropData: any[]): string | undefined {
    return airdropData.find(e => e.address === address)?.earnings;
}
```

Reply:
Done